collect_app/src/androidTest/java/org/odk/collect/android/regression/FillBlankFormTest.java

```diff
@@ -611,4 +620,160 @@ public void questions_shouldHavePrefilledValue() {
+                .checkBackNavigationButtonIsNotsDisplayed()
+                .checkNextNavigationButtonIsDisplayed()
+                .rotateToLandscape(new FormEntryPage("All widgets", rule))
+                .checkBackNavigationButtonIsNotsDisplayed()
+                .checkNextNavigationButtonIsDisplayed()
+                .rotateToPortrait(new FormEntryPage("All widgets", rule))
+                .checkBackNavigationButtonIsNotsDisplayed()
+                .checkNextNavigationButtonIsDisplayed();
+    }
+
+    @Test
+    public void groups_shouldBeVisibleInJumpScreen() {
+        //TestCase28
+        new MainMenuPage(rule)
+                .startBlankForm("nested-repeats-complex")
+                .swipeToNextQuestion()
+                .swipeToNextQuestion()
+                .clickOnAddGroup()
+                .inputText("La")
+                .swipeToNextQuestion()
+                .swipeToNextQuestion()
+                .clickOnAddGroup()
+                .inputText("Le")
+                .swipeToNextQuestion()
+                .clickOnAddGroup()
+                .inputText("Be")
+                .swipeToNextQuestion()
+                .clickOnDoNotAddGroup()
+                .clickOnDoNotAddGroup()
+                .clickOnAddGroup()
+                .inputText("Bu")
+                .swipeToNextQuestion()
+                .clickOnDoNotAddGroup()
+                .clickGoToArrow()
+                .clickOnText("Friends")
+                .checkListSizeInHierarchy(1)
+                .clickOnElementInHierarchy(0)
+                .clickOnText("Pets")
+                .checkListSizeInHierarchy(2)
+                .clickGoUpIcon()
+                .clickGoUpIcon()
+                .clickGoUpIcon()
+                .clickOnText("Enemies")
+                .checkListSizeInHierarchy(1);
+    }
+
+    @Test
+    public void hierachyView_shouldNoChangeAfterScreenRotation() {
+        //TestCase29
+        new MainMenuPage(rule)
+                .startBlankFormWithRepeatGroup("Repeat Group")
+                .clickOnDoNotAddGroup(new FormEntryPage("Repeat Group", rule))
+                .clickGoToArrow()
+                .clickGoUpIcon()
+                .checkIfElementInHierarchyMatchesToText("Group Name", 0)
+                .rotateToLandscape(new FormEntryPage("Repeat Group", rule))
+                .checkIfElementInHierarchyMatchesToText("Group Name", 0)
+                .rotateToPortrait(new FormEntryPage("Repeat Group", rule))
+                .checkIfElementInHierarchyMatchesToText("Group Name", 0);
+    }
+
+    @Test
+    public void hierarchyView_shouldBeDisplayed() {
+        //TestCase30
+        new MainMenuPage(rule)
+                .startBlankFormWithRepeatGroup("Repeat Group")
+                .clickOnDoNotAddGroup(new FormEntryPage("Repeat Group", rule))
+                .clickGoToArrow()
+                .clickJumpEndButton()
+                .checkIsStringDisplayed(R.string.quit_entry)
```

Comment:
It would be good to convert it to something like `isEndScreenVisible` or maybe I'm missing something?